libcxx/include/__algorithm/move_backward.h

```diff
@@ -86,7 +86,8 @@ struct __move_backward_impl {
   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX14 pair<_InIter, _OutIter>
   operator()(_InIter __first, _InIter __last, _OutIter __result) const {
     using _Traits = __segmented_iterator_traits<_OutIter>;
-    using _DiffT  = typename common_type<__iter_diff_t<_InIter>, __iter_diff_t<_OutIter> >::type;
```

Comment:
I think we should be using the `AlgPolicy` here to extract the difference type. That looks like a bug to me. That actually seems to be the case for all of the segmented iterator optimizations.